diff -ur others/libkorebot-1.11/Makefile libkorebot-1.11-kb1/Makefile
--- others/libkorebot-1.11/Makefile	2008-02-12 21:44:50.000000000 +0100
+++ libkorebot-1.11-kb1/Makefile	2008-02-12 21:52:10.000000000 +0100
@@ -54,6 +54,7 @@
 
 # Specify the target system. The corresponding sytem specify
 # section bellow should be edited as well.
+#TARGET_SYSTEM	= korebot-2.6
 TARGET_SYSTEM	= korebot
 #TARGET_SYSTEM	= x86
 
@@ -66,7 +67,7 @@
 # Default section (use TARGET variable) if you want to use gcc
 #---------------------------------------------------------------------
 
-ifneq ($(TARGET_SYSTEM),korebot)
+ifneq (,$(findstring korebot,$(TARGET_SYSTEM)))
 
 CC		= gcc
 LD		= ld
@@ -82,7 +83,14 @@
 # KoreBot specific section (use TARGET variable)
 #---------------------------------------------------------------------
 
-ifeq ($(TARGET_SYSTEM),korebot)
+ifeq ($(TARGET_SYSTEM),korebot-2.6)
+
+CC		= arm-angstrom-linux-gnueabi-gcc
+LD		= arm-angstrom-linux-gnueabi-ld
+AR		= arm-angstrom-linux-gnueabi-ar
+AS		= arm-angstrom-linux-gnueabi-as
+
+else
 
 CC		= arm-linux-gcc
 LD		= arm-linux-ld
diff -ur others/libkorebot-1.11/src/utils/kt_hexparser.c libkorebot-1.11-kb1/src/utils/kt_hexparser.c
--- others/libkorebot-1.11/src/utils/kt_hexparser.c	2005-10-25 13:45:34.000000000 +0200
+++ libkorebot-1.11-kb1/src/utils/kt_hexparser.c	2008-02-12 18:54:15.000000000 +0100
@@ -3,6 +3,7 @@
  * P.Bureau - 09.2005
  */
 #include "kt_hexparser.h"
+#include <string.h>
 
 size_t strlcpy(char *dst, const char *src, size_t siz)
 {
diff -ur others/libkorebot-1.11/src/utils/ucloadi2c.c libkorebot-1.11-kb1/src/utils/ucloadi2c.c
--- others/libkorebot-1.11/src/utils/ucloadi2c.c	2006-02-08 11:22:56.000000000 +0100
+++ libkorebot-1.11-kb1/src/utils/ucloadi2c.c	2008-02-12 18:53:42.000000000 +0100
@@ -2,6 +2,7 @@
 #include "i2ccom.h"
 #include<sys/types.h>
 #include<sys/time.h>
+#include <stdlib.h>
 
 /* Acknowledge message for the client */
 #define bl_ack_ack     'A'

