--- linux-2.6.24gum/arch/arm/mach-pxa/pxa27x.c	2008-01-24 23:58:37.000000000 +0100
+++ linux-2.6.24gum/arch/arm/mach-pxa/pxa27x_patched.c	2009-07-08 15:45:30.000000000 +0200
@@ -138,8 +138,11 @@
 
 	INIT_CKEN("UARTCLK", FFUART, 14857000, 1, &pxa_device_ffuart.dev),
 	INIT_CKEN("UARTCLK", BTUART, 14857000, 1, &pxa_device_btuart.dev),
+	INIT_CKEN("UARTCLK", STUART, 14745600, 1, &pxa_device_stuart.dev),
+/* BT error: removed JT 20090708
+  http://www.nabble.com/Bluetooth-fix-for-OE-318%2C-kernel-2.6.24-on-Verdex-td22181590.html#a22181590
 	INIT_CKEN("UARTCLK", STUART, 14857000, 1, NULL),
-
+*/
 	INIT_CKEN("I2SCLK",  I2S,  14682000, 0, &pxa_device_i2s.dev),
 	INIT_CKEN("I2CCLK",  I2C,  32842000, 0, &pxa_device_i2c.dev),
 	INIT_CKEN("UDCCLK",  USB,  48000000, 5, &pxa_device_udc.dev),
 	
 	

