commit 1cfafd8020824892c5f18a77d8680fec5e7efea6
Author: Ryan Hunter <rghunter@bu.edu>
Date:   Sat Jun 2 19:49:42 2012 -0400

    changed overo environment variables to include mem

diff --git a/include/configs/omap3_overo.h b/include/configs/omap3_overo.h
index 8114194..54666de 100644
--- a/include/configs/omap3_overo.h
+++ b/include/configs/omap3_overo.h
@@ -166,6 +166,7 @@
 	"mmcrootfstype=ext3 rootwait\0" \
 	"nandroot=ubi0:rootfs ubi.mtd=4\0" \
 	"nandrootfstype=ubifs\0" \
+	"mem=mem=96M@0x80000000 mem=128M@0x88000000\0" \
 	"mmcargs=setenv bootargs console=${console} " \
 		"${optargs} " \
 		"mpurate=${mpurate} " \
@@ -178,6 +179,7 @@
 		"${optargs} " \
 		"mpurate=${mpurate} " \
 		"vram=${vram} " \
+		"${mem} " \
 		"omapfb.mode=dvi:${dvimode} " \
 		"omapdss.def_disp=${defaultdisplay} " \
 		"root=${nandroot} " \
