commit 60f549b2b5035b29bc0e4c1c1b5181809f321052
Author: Ryan Hunter <rghunter@bu.edu>
Date:   Sun Jun 3 04:17:25 2012 -0400

    fix for cmemk

diff --git a/include/configs/omap3_overo.h b/include/configs/omap3_overo.h
index a0a7a1c..e5d9511 100644
--- a/include/configs/omap3_overo.h
+++ b/include/configs/omap3_overo.h
@@ -159,10 +159,12 @@
 	"mmcrootfstype=ext3 rootwait\0" \
 	"nandroot=ubi0:rootfs ubi.mtd=4\0" \
 	"nandrootfstype=ubifs\0" \
+	"mem=mem=96M@0x80000000 mem=128M@0x88000000\0" \
 	"mmcargs=setenv bootargs console=${console} " \
 		"${optargs} " \
 		"mpurate=${mpurate} " \
 		"vram=${vram} " \
+		"${mem} " \
 		"omapfb.mode=dvi:${dvimode} " \
 		"omapdss.def_disp=${defaultdisplay} " \
 		"root=${mmcroot} " \
@@ -171,6 +173,7 @@
 		"${optargs} " \
 		"mpurate=${mpurate} " \
 		"vram=${vram} " \
+		"${mem} " \
 		"omapfb.mode=dvi:${dvimode} " \
 		"omapdss.def_disp=${defaultdisplay} " \
 		"root=${nandroot} " \
