Index: linux-2.6.22gum/drivers/pcmcia/pxa2xx_gumstix.c
===================================================================
--- linux-2.6.22gum.org/drivers/pcmcia/pxa2xx_gumstix.c	2008-09-11 08:33:42.000000000 +0200
+++ linux-2.6.22gum/drivers/pcmcia/pxa2xx_gumstix.c	2008-09-11 08:57:07.000000000 +0200
@@ -30,6 +30,8 @@
 
 #include "soc_common.h"
 
+#define KOREBOT2 1
+
 static struct pcmcia_irqs gumstix_pcmcia_irqs0[] = {
 	{ 0, GUMSTIX_S0_nCD_IRQ,	"CF0 nCD"     },
 	{ 0, GUMSTIX_S0_nSTSCHG_IRQ,	"CF0 nSTSCHG" },
@@ -214,6 +216,12 @@
 static void __init gumstix_check_if_netCF_vx(void)
 {
 	void *network_controller_memory = ioremap(0x04000300,16);
+
+	#ifdef KOREBOT2
+	net_cf_vx_mode = 1; // the Korebot2 uses the same GPIO as the netCF_VX
+  printk("CF Korebot 2\n");
+
+	#else	
 	// Look for the special 91c111 value in the bank select register
 	if((0xff00 & readw(network_controller_memory+0x0e)) == 0x3300) {
 		printk("Detected netCF-vx board: using older GPIO configuration\n");
@@ -222,6 +230,8 @@
 		printk("Not netCF-vx board: using newer GPIO configuration\n");
 		net_cf_vx_mode = 0;
 	}
+	#endif
+
 	iounmap(network_controller_memory);
 }
 #endif
